### PR TITLE
Adds `eslint-plugin-eslint-comments`

### PR DESCRIPTION
**Pull request by cartogram**
Closes: https://github.com/Shopify/eslint-plugin-shopify/issues/88

### CODE REVIEW

File: lib/config/rules/eslint-comments.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+// see https://github.com/mysticatea/eslint-plugin-eslint-comments
+
+module.exports = {
```
Comment (GoodForOneFare) [inline lib/config/rules/eslint-comments.js:3]: Missing [`disable-enable-pair`](https://github.com/mysticatea/eslint-plugin-eslint-comments/blob/master/docs/rules/disable-enable-pair.md).

File: lib/config/rules/eslint-comments.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+// see https://github.com/mysticatea/eslint-plugin-eslint-comments
```
Comment (GoodForOneFare) [inline lib/config/rules/eslint-comments.js]: The docs for this repo aren't great.  I'd prefer a link to [the page that lists the rules](https://github.com/mysticatea/eslint-plugin-eslint-comments/tree/master/docs/rules#best-practices).

**Review by GoodForOneFare — APPROVED** [review]
Looks great  👍 I agree with the disabled rules.  Thanks for taking care of this!

File: package.json [hunk 3]
```diff
@@ -84,6 +84,7 @@
     "eslint-plugin-ava": "5.1.0",
     "eslint-plugin-babel": "5.1.0",
     "eslint-plugin-chai-expect": "1.1.1",
+    "eslint-plugin-eslint-comments": "^3.0.1",
```
Comment (kaelig) [inline package.json]: The other dependencies are pinned to a specific version, maybe this one should be as well?

**Comment by cartogram** [comment]
@GoodForOneFare made the requested changes, but the `config:all` test is failing and I am lost as to why. Do you know how I would go about debugging this? This is failing in a few of my [recent PRs](https://github.com/Shopify/eslint-plugin-shopify/pull/149).

**Comment by GoodForOneFare** [comment]
The test is failing because eslint-plugin-shopify's test files violate the new rules.

```
➜  eslint-plugin-shopify git:(add-comments) node_modules/.bin/eslint --config /Users/gord/shopify/eslint-plugin-shopify/lib/config/all.js /Users/gord/shopify/eslint-plugin-shopify/tests/lib/config
Warning: React version not specified in eslint-plugin-react settings. See https://github.com/yannickcr/eslint-plugin-react#configuration.

/Users/gord/shopify/eslint-plugin-shopify/tests/lib/config/all.js
  2:19  error  Requires 'eslint-enable' directive for 'flowtype/require-valid-file-annotation'  eslint-comments/disable-enable-pair

/Users/gord/shopify/eslint-plugin-shopify/tests/lib/config/prettier.js
   2:19  error  Requires 'eslint-enable' directive for 'flowtype/require-valid-file-annotation'  eslint-comments/disable-enable-pair
   6:29  error  'shopify/strict-component-boundaries' rule is disabled but never reported        eslint-comments/no-unused-disable
  17:37  error  'function-paren-newline' rule is disabled but never reported                     eslint-comments/no-unused-disable

/Users/gord/shopify/eslint-plugin-shopify/tests/lib/config/typescript-prettier.js
   2:19  error  Requires 'eslint-enable' directive for 'flowtype/require-valid-file-annotation'  eslint-comments/disable-enable-pair
   6:29  error  'shopify/strict-component-boundaries' rule is disabled but never reported        eslint-comments/no-unused-disable
  19:37  error  'function-paren-newline' rule is disabled but never reported                     eslint-comments/no-unused-disable

✖ 7 problems (7 errors, 0 warnings)
```

The `all` test was intended more as a, "are all config files required, and are they valid JS?" check.  So failing on lint errors is weird.

**Comment by BPScott** [comment]
Somewhat late to the party but can we use `"eslint-comments/disable-enable-pair": ["error", {"allowWholeFile": true}]` to allow for disabling a rule for the whole file. It's often useful in scripts to disable the no-console rule across a whole file for instance

https://github.com/mysticatea/eslint-plugin-eslint-comments/blob/master/docs/rules/disable-enable-pair.md#options

**Comment by cartogram** [comment]
@BPScott can you make an issue and assign it to me.

**Comment by BPScott** [comment]
Done #183

**Pull request merged**